### PR TITLE
top_file gadgets shows complete file path now

### PR DESCRIPTION
**Pull request by h4l0gen**
# [Title: describe the change in one sentence]

`top_file` gadget now shows absolute path in `FILE` parameter.

## How to use

run top_file gadget, and create a pod and put some data into a file.

## Testing done
I tested it on minikube locally. First I tried it to see what changes we require.  when I did this `echo "Hello, World!" > /tmp/testfile.txt`, `top_file` shows filename as `testfile.txt`. After my changes I ran same command and filepath shown is `tmp/testfile.txt`

fixes: #3234

### CODE REVIEW

**Comment by h4l0gen** [comment]
Here I am attaching screenshot, 

Screenshot from 2024-08-03 12-05-44

File: pkg/gadgets/top/file/tracer/bpf/filetop.bpf.c [hunk 1]
```diff
@@ -23,10 +24,18 @@ struct {
...
+	bpf_probe_read(&f_path, sizeof(struct path), &file->f_path);
+
+	char *path_str = get_path_str(&f_path);
+	if(path_str) {
+		bpf_probe_read_str(buf, PATH_MAX, path_str);
+	} else {
+		struct qstr dname;
+		dname = BPF_CORE_READ(file, f_path.dentry, d_name);
+		bpf_probe_read_kernel(buf, size, dname.name);
+	}
```
Comment (mauriciovasquezbernal) [inline pkg/gadgets/top/file/tracer/bpf/filetop.bpf.c:37]: What's the reason to have this fallback?

**Review by mauriciovasquezbernal — COMMENTED** [review]
The change looks hook, however we don't want to update the gadgets under pkg/gadgets anymore. Would you mind applying this change to ` gadgets/top_file/` only? Thanks!

**Comment by mauriciovasquezbernal** [comment]
Closing in favor of #3347.

**Pull request closed without merging**